src/main/java/com/google/devtools/build/lib/bazel/repository/cache/RepoContentsCache.java

```diff
@@ -58,6 +86,15 @@ private static CandidateRepo fromRecordedInputsFile(Path recordedInputsFile) {
       return new CandidateRepo(
           recordedInputsFile, recordedInputsFile.replaceName(contentsDirBaseName));
     }
+
+    /** Updates the mtime of the recorded inputs file, to delay GC for this entry. */
+    public void touch() {
+      try {
+        recordedInputsFile.setLastModifiedTime(Instant.now().toEpochMilli());
```

Comment:
I don't know how bad the clock skew can be, but a regular `touch` would do the equivalent of `recordedInputsFile.setLastModifiedTime(Path.NOW_SENTINEL_TIME);`.

Maybe it's safer to go with that and also obtain the current filesystem time below when comparing for age?

Reply:
Is the repository cache per-machine or per-user? if it's per-machine (and shareable between users) you *must* use `NOW_SENTINEL_TIME`, as the the kernel will not let you set the mtime on a file owned by another user (even if you have write permission for it).

Reply:
> also obtain the current filesystem time below when comparing for age?

How does one do that? :) I couldn't find a "companion API" for `Path.NOW_SENTINEL_TIME`.

Reply:
You could just touch a temporary file and read it back.

Reply:
Done.